### PR TITLE
GF-56, GF-61 - Authentication with devise token auth and Redux persist

### PR DESCRIPTION
**Pull request by MIKALAKE**

### CODE REVIEW

File: src/process/api/index.js
Comment (iamdavidpetric) [inline src/process/api/index.js]: put the base url here

File: src/process/slices/userSlice.js
Comment (iamdavidpetric) [inline src/process/slices/userSlice.js]: no console.log in PR

File: src/process/slices/userSlice.js
Comment (iamdavidpetric) [inline src/process/slices/userSlice.js]: you don;t use state and payload also want to return only initial state(define it above userSlice)

File: src/process/api/index.js
Comment (MIKALAKE) [reply src/process/api/index.js]: done

File: src/process/slices/userSlice.js
Comment (MIKALAKE) [reply src/process/slices/userSlice.js]: done

File: src/process/slices/userSlice.js
Comment (MIKALAKE) [reply src/process/slices/userSlice.js]: done

File: src/process/api/index.js
Comment (iamalextodoran) [inline src/process/api/index.js]: I missed a console log?!

File: src/process/slices/userSlice.js
Comment (iamalextodoran) [inline src/process/slices/userSlice.js]: @iamdavidpetric anything else to be done in here?

File: src/process/slices/userSlice.js
Comment (iamalextodoran) [inline src/process/slices/userSlice.js]: @iamdavidpetric or here?

File: src/process/slices/userSlice.js
Comment (iamalextodoran) [inline src/process/slices/userSlice.js]: @iamdavidpetric or here?

File: src/process/slices/userSlice.js
Comment (iamalextodoran) [inline src/process/slices/userSlice.js]: @iamdavidpetric or here?

File: src/show/components/Navbar/index.jsx
Comment (iamalextodoran) [inline src/show/components/Navbar/index.jsx]: @iamdavidpetric or here?

File: src/show/pages/public/Home/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.jsx]: Why the empty line?

File: src/show/pages/public/LogIn/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/LogIn/index.jsx:8]: Why the empty line before this?

File: src/show/pages/public/LogIn/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/LogIn/index.jsx]: Could fit in one line. Also why name them like this?

File: src/show/pages/public/LogIn/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/LogIn/index.jsx]:
```suggestion
    dispatch(logIn(client));
```
@iamdavidpetric neither this one?

File: src/show/pages/public/Register/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/Register/index.jsx:7]: No empty line before

File: src/show/pages/public/Register/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/Register/index.jsx]: Interesting name again

File: src/show/pages/public/Register/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/Register/index.jsx]: Client is already an object containing all of those you need, don't bother doing them like this when you can send the whole object.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/process/api/index.js
Comment (MIKALAKE) [reply src/process/api/index.js]: done

File: src/process/slices/userSlice.js
Comment (MIKALAKE) [reply src/process/slices/userSlice.js]: done

File: src/process/slices/userSlice.js
Comment (MIKALAKE) [reply src/process/slices/userSlice.js]: @iamalextodoran ?

File: src/process/slices/userSlice.js
Comment (MIKALAKE) [reply src/process/slices/userSlice.js]: done

File: src/process/slices/userSlice.js
Comment (MIKALAKE) [reply src/process/slices/userSlice.js]: done

File: src/show/components/Navbar/index.jsx
Comment (MIKALAKE) [reply src/show/components/Navbar/index.jsx]: done

File: src/show/pages/public/Home/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/Home/index.jsx]: done

File: src/show/pages/public/LogIn/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/LogIn/index.jsx:8]: done

File: src/show/pages/public/LogIn/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/LogIn/index.jsx]: done

File: src/show/pages/public/LogIn/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/LogIn/index.jsx]: done

File: src/show/pages/public/Register/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/Register/index.jsx:7]: done

File: src/show/pages/public/Register/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/Register/index.jsx]: done

File: src/show/pages/public/Register/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/Register/index.jsx]: done

File: src/process/slices/userSlice.js
Comment (iamalextodoran) [reply src/process/slices/userSlice.js]: You don't have anything to return

File: src/show/pages/public/Register/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/Register/index.jsx]: Does this work?

File: src/show/pages/public/Register/index.jsx
Comment (iamalextodoran) [inline src/show/pages/public/Register/index.jsx]: Instead of this you should be building the user object dynamically by passing the label (e.g. first_name, last_name, email etc.) and the value. 

const onFieldChange = (key, value) => {
... build here
}

so you can use it like this

onChange={e => onFieldChange('first_name', e.target.value)} - for the first_name text field

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/Application.js
Comment (iamalextodoran) [inline src/Application.js:5]: This is grouped with the above. It's an absolute import

File: src/index.js
Comment (iamalextodoran) [inline src/index.js:8]: This is grouped separaely since it's a relative import.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/Application.js
Comment (MIKALAKE) [reply src/Application.js:5]: done

File: src/index.js
Comment (MIKALAKE) [reply src/index.js:8]: done

File: src/show/pages/public/Register/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/Register/index.jsx]: Yes I know it works, should not ?

File: src/show/pages/public/Register/index.jsx
Comment (MIKALAKE) [reply src/show/pages/public/Register/index.jsx]: done

File: src/index.js
Comment (iamalextodoran) [inline src/index.js]: Last comments on this PR, if you solve them just merge this, don't wait for my approval.

We group those by: 
* external libraries
* absolute paths
* relative paths

Think about this, do the proper thing.

File: src/process/slices/userSlice.js
Comment (iamalextodoran) [inline src/process/slices/userSlice.js]: `payload` doesn't seem to be used.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

**Pull request merged**
